schema.json

```diff
@@ -542,8 +510,67 @@
           "type": "string"
         }
       },
-      "required": ["A01"],
-      "additionalProperties": false
+      "required": ["A01", "A02", "A03", "A04", "A05", "A07"]
+    },
+    "CommercialTransactionReceivers": {
+      "title": "CommercialTransactionReceivers",
+      "description": "",
+      "oneOf": [
+        {
+          "properties": {
+            "A06": {
+              "allOf": [
+                {
+                  "$ref": "#/definitions/Company"
+                }
+              ],
+              "description": "The purchaser of the product and receiver of the certificate"
+            }
+          },
+          "required": ["A06"]
+        },
+        {
+          "properties": {
+            "A06.1": {
+              "allOf": [
+                {
+                  "$ref": "#/definitions/Company"
+                }
+              ],
+              "description": "The purchaser of the product"
+            },
+            "A06.2": {
+              "allOf": [
+                {
+                  "$ref": "#/definitions/Company"
+                }
+              ],
+              "description": "The consignee of the product"
+            },
+            "A06.3": {
+              "allOf": [
+                {
+                  "$ref": "#/definitions/Company"
+                }
+              ],
+              "description": "The receiver/consignee of the certificate"
+            }
+          },
+          "required": ["A06.1"]
+        }
+      ]
+    },
+    "CommercialTransaction": {
+      "title": "CommercialTransaction",
+      "description": "List parties and details involved in the transaction",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CommercialTransactionBase"
+        },
+        {
+          "$ref": "#/definitions/CommercialTransactionReceivers"
+        }
+      ]
```

Comment:
In `CommercialTransactionBase` are defined all the properties except those related to "receivers" (A06, A06.1, A06.2 and A06.3)

In `CommercialTransactionReceivers` (we could pick a better name, i just didn't find it ), translate this rule:
`Either A06 or A06.1
If A06.1 then .2 and .3 are optional.`

The drawback with that approach, is that we cannot set `additionalProperties = false` anymore in `CommercialTransaction`.

I tested that schema to validate some example and it works.